src/core/pipeline/pipeline.js

```diff
@@ -0,0 +1,35 @@
+export default class Pipeline {
```

Comment:
1. I'm not sure how you will do it for async operations, but defenetly we can put try catch, and invoke the last(system finish) pipe?
2. You mean cancel, good point? if we are going to provide cancelation in context, next shoud be there to
(data, ctx) => if something wrokng { ctx.cancel() ??